test/canary/canary_test.go

```diff
@@ -32,46 +30,83 @@ func init() {
 	environment.RegisterEnvironmentMetaDataFlags(envMetaDataStrings)
 }
 
-// This copies the canary config, starts the agent, and confirm version is correct.
-// Sanity test runs earlier in the terraform code execution
+// TestCanary verifies customers can download the agent, install it, and it "works".
+// Reports metrics on the number of download failures, install failures, and start failures.
 func TestCanary(t *testing.T) {
-	// Canary set up
-	downloadVersionFile()
+	installerFilePath := "./downloaded_cwa.rpm"
+	err := downloadInstaller(installerFilePath)
+	reportMetric(t, "DownloadFail", err)
```

Comment:
It would be better to leverage custom metrics we have (e.g statsd)

Reply:
I don't think that is better. If there is an issue with downloading, installing, or starting CloudWatch Agent then we cannot rely on sending statsd metrics to CloudWatch Agent as the mechanism to report such failures. We intentionally want to report metrics using AWS SDK, or AWS CLI, or something other than CWA itself.

Reply:
I think that's what we'd need to do. Right now the `DownloadFail = 1`, then none of the other metrics will get reported, so we'd need to count those as failures as well.

Reply:
We'd need to have some buffer for missing data during host replacements.

Reply:
@SaxyPandaBear yes we will continue to alarm if infrastructure metrics are missing data. As @jefchien said, there is be an allowance for a few minutes of missing data due to the expected host replacement once a day.

The custom metrics like "DownloadFail" might not be necessary. I think they will make it easier to tell what exactly went wrong. And eventually we may want many canary hosts in different regions or running different operating systems. In which case a single alarm on the "DownloadFail" metric (no dimensions) would suffice. We would still want some way to alarm if the agent on 1 of the many canary hosts was missing data though...